FirebaseMLModelDownloader/Sources/ModelDownloadTask.swift

```diff
@@ -26,34 +27,46 @@ class DownloadHandlers {
   typealias ProgressHandler = (Float) -> Void
   typealias Completion = (Result<CustomModel, DownloadError>) -> Void
 
+  var runsOnMainThread: Bool
```

Comment:
Hmm, is there a reason why we would want the completion handler to ever get called on the background thread? The download itself will always be in the background, I'm not sure if it's needed to have the extra complexity here.